uefi/section.go

```diff
@@ -169,8 +196,39 @@ func NewSection(buf []byte, fileOrder int) (*Section, error) {
 	// Slice buffer to the correct size.
 	s.buf = buf[:s.Header.ExtendedSize]
 
-	// Get the name.
-	if s.Header.Type == SectionTypeUserInterface {
+	// Section type specific data
+	switch s.Header.Type {
+	case SectionTypeGUIDDefined:
+		typeSpec := &SectionGUIDDefined{}
+		if err := binary.Read(r, binary.LittleEndian, typeSpec); err != nil {
+			return nil, err
+		}
+		s.TypeSpecific = &typeSpec
+
+		// Determine whether the encapsulated section needs processing.
+		if typeSpec.Attributes&uint16(GUIDEDSectionProcessingRequired) == 0 {
+			// No processing required
+			var err error
+			s.Encapsulated, err = NewSection(buf[typeSpec.DataOffset:], 0)
```

Comment:
If I'm right, and there can be multiple sections, this will need to mirror the looping construction behavior in File.